plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,648 @@
+In this case, the exit would be blocked entirely and the challenging user would receive `exit bond`.
+
+Option (1) checks that a transaction is exitable when the exit is started.
+This has lower communication cost and complexity but higher up-front gas cost.
+This option also ensures that only a single exit on any given transaction can exist at any point in time.
+Option (2) allows a user to assert that a transaction is exitable, but leaves the proof to a challenge-response game.
+This is cheaper up-front but adds complexity.
+This option must permit multiple exits on the same transaction, as some exits may provide invalid signatures.
+
+These are not the only possible mechanisms that prove a transaction is exitable.
+There may be further ways to optimize these two options.
+
+We still need to provide a deterministic ordering of exits by some priority.
+MoreVP exits are given a priority based on the position in the Plasma chain of the most recently included (youngest) input to that transaction.
+Unlike the MVP protocol, we give each input and output to a transaction the same priority.
+This should be implemented by inserting a single “exit” object into a priority queue of exits and tracking a list of inputs or outputs to be paid out once the exit is processed.
+
+
+#### Proving Canonicity
+
+Whether unspent inputs or unspent outputs are paid out in an exit depends on the canonicity of the referenced transaction.
+Unfortunately it’s too expensive to directly prove that a transaction is or is not canonical.
+Instead, we assume that the referenced transaction is canonical by default and allow a series of challenges and responses to determine the true canonicity of the transaction.
+
+The process of determining canonicity involves a challenge-response game.
+In the first period of the exit, any user may reveal a competing transaction that potentially makes the exiting transaction non-canonical.
+This competing transaction must be exitable, must share an input with the exiting transaction, and must be included in the Plasma chain.
+Multiple competing transactions can be revealed during this period, but only the oldest presented transaction is considered for the purposes of a response.
+
+If any transactions have been presented during the first period, any other user can respond to the challenge by proving that the exiting transaction is actually included in the chain before the oldest presented competing transaction.
+If this response is given, then the exiting transaction is determined to be canonical and the responder receives the `exit bond` placed by the user who started the exit.
+Otherwise, the exiting transaction is determined to be non-canonical and the challenger receives `exit bond`.
+
+Note that this challenge means it’s possible for an honest user to lose `exit bond` as they might not be aware their transaction is non-canonical.
+We address this attack vector and several mitigations in detail later.
+
+
+#### Joining an Exit
```

Comment:
ditto, not ok with having two terms for that. Let's just use "piggyback" everywhere (or if you'd like to reconsider, let's forget piggybacking and just join everywhere, but that might be hard). Multiple occurrences of that here. Paragraphs like [this](https://github.com/omisego/research/pull/44/files#diff-056213b9987c8cd9d93d6615e132221fR213) that use both will become confusing if they aren't already

Reply:
Replacing all occurrences of "join" with "piggyback."